lab/lablib/SoundManager.js

```diff
@@ -171,6 +180,51 @@ export class SoundManager {
     );
   }
 
+  /**
+   * Get the current value of a parameter from the currently playing score
+   * @param {string} param_id The ID of the parameter in the score
+   * @returns {number | undefined} The current value of the parameter, or undefined if no value is set
+   */
+  get_param(param_id) {
+    const tween_list = this.score_tweens[this?.current_score]?.[param_id];
+    if (tween_list === undefined || tween_list.length === 0) {
+      return undefined;
+    }
+
+    // TODO: This would be a good case for binary search
+    const time = this.transport_time;
+
+    let tween;
+    if (time < tween_list[0].start_time) {
+      tween = tween_list[0];
+    } else if (time >= tween_list[tween_list.length - 1].end_time) {
+      tween = tween_list[tween_list.length - 1];
+    } else {
+      tween = tween_list.find(
+        (tween) => time >= tween.start_time && time < tween.end_time
+      );
+    }
+
+    return tween.get_value(time);
+  }
+
+  /**
+   * Convert a timeline of ParamCurve to an array of Tween, assuming a start time of 0
+   * @param {import("./music/Timeline.js").Timeline<ParamCurve>} timeline the timeline to convert
+   * @return {Tween[]} The corresponding tweens,
+   */
+  compile_tweens(timeline) {
```

Comment:
Write down a note that maybe we need a `CompiledScore` object so SoundManager doesn't get bloated